samza-yarn/src/main/java/org/apache/samza/config/YarnConfig.java

```diff
@@ -31,6 +31,14 @@
    */
   public static final String PACKAGE_PATH = "yarn.package.path";
 
+  /**
+   * Type of Samza application submitted to YARN
+   * Can be used to distinguish between different job types/APIs (eg: HighLevel, LowLevel, SQL, Beam etc)
+   */
+  public static final String YARN_APPLICATION_TYPE = "yarn.application.type";
+  public static final String DEFAULT_SAMZA_APPLICATION_TYPE = "Samza";
```

Comment:
Consider linking a JIRA with some background on this config?

Reply:
also, maybe add how you envision this being used?

Reply:
@vjagadish1989 updated the PR with the ticket